src/Rings.jl

```diff
@@ -241,3 +241,49 @@ function is_finite(R::NCRing)
   c == 1 && return true
   throw(NotImplementedError(:is_finite, R))
 end
+
+@doc raw"""
+    is_noetherian(R::Ring)
+
+Check if the ring $R$ is Noetherian.
```

Comment:
We also want this for modules in the future. So perhaps:
```suggestion
    is_noetherian(A::Ring)
    is_noetherian(A::Module)

Check if $A$ is Noetherian.
```